src/renderers/dom/shared/__tests__/ReactDOMComponent-test.js

```diff
@@ -791,7 +791,40 @@ describe('ReactDOMComponent', () => {
       );
     });
 
-    it('should emit warning when using shady dom with a custom component', () => {
+    it('should emit a warning once for a named custom component using shady DOM', () => {
+      if (ReactDOMFeatureFlags.useCreateElement) {
+        spyOn(console, 'error');
+
+        var defaultCreateElement = document.createElement.bind(document);
+
+        try {
+          document.createElement = element => {
+            var container = defaultCreateElement(element);
```

Comment:
Nit: this is not a container per se, it's just a DOM node. We call the topmost DOM node "container" because it contains the whole React app. Let's rename this to `node`.